File: clients/swift/src/RateLimiting/RateLimitState.swift
import Foundation

/// Per-policy quota state from the `RateLimit` response header.
public struct RateLimitState: Sendable, Equatable {
    public let name: String
    public let remaining: Int
    public let resetSeconds: Int

    public init(name: String, remaining: Int, resetSeconds: Int) {
        self.name = name
        self.remaining = remaining
        self.resetSeconds = resetSeconds
    }
}

File: clients/csharp/src/RateLimiting/RateLimitState.cs
using System.Text.Json.Serialization;

namespace MyMind.RateLimiting;

/// <summary>Per-policy quota state from the <c>RateLimit</c> response header.</summary>
public class RateLimitState
{
    [JsonPropertyName("name")]
    public required string Name { get; init; }

    [JsonPropertyName("r")]
    public required int Remaining { get; init; }

    [JsonPropertyName("t")]
    public required int ResetSeconds { get; init; }
}
